### PR TITLE
Workbench Hotfixes

### PR DESCRIPTION
**Pull request by Paneedah**
## 📝 Description

This should now finally properly track items within the workbench and remove them without any weird issues (such as not being able to craft anything).

## 🚦 Testing 

Generic testing done locally in staffchat, boring. This does however need to be messed with and tested further.

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📑 Documentation
- [ ] 📓 Javadoc
- [ ] 🍕 Comments
- [x] 🙅 No documentation needed

## Summary by CodeRabbit

- **Refactor**
	- Improved the logic for item requirements during crafting selection for a more intuitive user experience.
- **Bug Fixes**
	- Enhanced the verification and removal process of items in crafting recipes to ensure accuracy and reliability.

### CODE REVIEW

**Comment by Paneedah** [comment]
Warning: OreDict support was removed in this PR. Awaiting for @SticksDeveloper to re-implement.

**Comment by Desoroxxx** [comment]
> Warning: OreDict support was removed in this PR. Awaiting for @SticksDeveloper to re-implement.

OreDict is not removed in this PR, and it was never supported, a fix for OreDict related crash is removed in this pr this will be fixed shortly

**Comment by SticksDeveloper** [comment]
It is good for Merge!

**Comment by SticksDeveloper** [comment]
Wrong button

File: src/main/java/com/paneedah/weaponlib/network/packets/StationPacket.java [hunk 1]
```diff
@@ -26,6 +25,7 @@
 import net.minecraftforge.fml.common.network.simpleimpl.MessageContext;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.oredict.OreDictionary;
+import org.lwjgl.Sys;
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/network/packets/StationPacket.java:28]: Wtf

Comment (SticksDeveloper) [reply src/main/java/com/paneedah/weaponlib/network/packets/StationPacket.java:28]: Wasn't me this time

**Pull request closed without merging**